File: String/LongestCommonPrefix.js
/**
 * 最长公共前缀
 * source: https://leetcode-cn.com/explore/featured/card/top-interview-questions-easy/5/strings/40/
 */

var longestCommonPrefix = function(strs) {
	let index=0;
	let commonPrefix = '';
	while(strs.length > 0) {
		let currentChar = strs[0].charAt(index);
		if (currentChar == '') return commonPrefix;

		for (var i = 1; i < strs.length; i++) {
			let nextChar = strs[i].charAt(index);
			if (nextChar == '' || currentChar != nextChar) {
				return commonPrefix;
			} 
		}

		commonPrefix += currentChar;
		index++;
	}

	return commonPrefix;
};

let strs = [''];
console.log(longestCommonPrefix(strs));

File: 字节跳动/longestCommonPrefix.swift
/**
 * 公共最长前缀
 * source: https://leetcode-cn.com/explore/interview/card/bytedance/242/string/1014/
 */

import Foundation

class Solution {
    func longestCommonPrefix(_ strs: [String]) -> String {
        guard strs.count > 0 else { return "" }
        
        var commonPrefit = String()
        let firstStrArray = Array(strs[0])
        
        for char in firstStrArray {
            commonPrefit.append(char)
            for strIndex in 1..<strs.count {
                let currentStr: String = strs[strIndex]
                if (!currentStr.hasPrefix(commonPrefit)) {
                    commonPrefit.removeLast();
                    return commonPrefit
                }
            }
        }
        return commonPrefit
    }
}


